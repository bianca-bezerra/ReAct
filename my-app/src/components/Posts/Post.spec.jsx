import { render, screen } from '@testing-library/react'
import { Posts } from '.'

const props = {
  posts: [
    {
      id: "1",
      title: "titulo 1",
      body: "corpo do post 1",
      cover: "fotinha 1",
      alt: 'alt de fotinha 1'
    },
    {
      id: "2",
      title: "titulo 2",
      body: "corpo do post 2",
      cover: "fotinha 2",
      alt: 'alt de fotinha 2'
    },
    {
      id: "3",
      title: "titulo 3",
      body: "corpo do post 3",
      cover: "fotinha 3",
      alt: 'alt de fotinha 3'
    }
  ]
}

const testAllByRoleElements = (type, nameExpected, length) => {
  return expect(screen.getAllByRole(type, { name: new RegExp(nameExpected, "i") }))
    .toHaveLength(length)
}

describe('<Posts/>', () => {
  it('should render posts', () => {
    render(<Posts {...props} />)
    testAllByRoleElements("heading", "titulo", 3)
    testAllByRoleElements("img", "fotinha", 3)
  })

  it('should have posts', () => {
    render(<Posts />)
    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
