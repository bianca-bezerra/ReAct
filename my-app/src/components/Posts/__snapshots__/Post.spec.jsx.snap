// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <div
      class="post-content"
    >
      <img
        alt="alt de fotinha 1"
        src="fotinha 1"
      />
      <h1>
        titulo 1
      </h1>
      <p />
    </div>
  </div>
  <div
    class="post"
  >
    <div
      class="post-content"
    >
      <img
        alt="alt de fotinha 2"
        src="fotinha 2"
      />
      <h1>
        titulo 2
      </h1>
      <p />
    </div>
  </div>
  <div
    class="post"
  >
    <div
      class="post-content"
    >
      <img
        alt="alt de fotinha 3"
        src="fotinha 3"
      />
      <h1>
        titulo 3
      </h1>
      <p />
    </div>
  </div>
</div>
`;
